core/mm.c

```diff
@@ -176,14 +177,22 @@ static void map_os_memory(void)
 	list_for_each_entry(region, &mem_list, list) {
 		type = memory_region_type(region);
 		if ((type != MEMORY_REGION_TYPE_NORMAL) &&
-				(type != MEMORY_REGION_TYPE_DMA))
+		    (type != MEMORY_REGION_TYPE_DMA) &&
+		    /*
+		     * Map the memory in VM region, because we may touch the
+		     * kernel and fdt.
+		     */
+		    (type != MEMORY_REGION_TYPE_VM))
 			continue;
 
 		if (type == MEMORY_REGION_TYPE_DMA)
 			flags |= VM_IO;
 		else
 			flags |= VM_NORMAL;
```

Comment:
Is it appropriate to treat `VM` and `ramdisk` regions as `VM_NORMAL`?

Reply:
yes, need add MEMORY_REGION_TYPE_RAMDISK.